kpatch-build/kpatch-elf.c

```diff
@@ -76,6 +76,65 @@ int is_debug_section(struct section *sec)
 	       !strncmp(name, ".eh_frame", 9);
 }
 
+#ifdef __powerpc__
+
+/* Symbol st_others value for powerpc */
+#define STO_PPC64_LOCAL_BIT     5
+#define STO_PPC64_LOCAL_MASK    (7 << STO_PPC64_LOCAL_BIT)
+#define PPC64_LOCAL_ENTRY_OFFSET(other)                                 \
+	        (((1 << (((other) & STO_PPC64_LOCAL_MASK) >> STO_PPC64_LOCAL_BIT)) >> 2) << 2)
+
+/*
+ * As per ELF ABI v2 specification, any function those symbol reference
+ * is to be dynamically resolved. The st_shndx member of such symbol
+ * is set to SHN_UNDEF. This function return 1, if the function's
+ * is local by checking for it's sym.st_shndx value.
+ */
+static int is_local_func(struct symbol *sym)
+{
+	if (sym->type == STT_FUNC && sym->sym.st_shndx == SHN_UNDEF)
+		return 0;
```

Comment:
Also, I think this function should be removed and this check should just be moved directly into `is_localentry_sym()`.  The function comment above doesn't seem to be very helpful anyway.